packages/client/hmi-client/src/components/Modal.vue

```diff
@@ -0,0 +1,81 @@
+<script setup lang="ts">
+/**
+ * A modal with content slots for a header, body, and footer. Use v-if to control visiblity.
+ * @example
+ * <modal v-if="isModalVisible">
+ * 		<template #header>Header content</template>
+ * 		<template #default>Body content</template>
+ * 		<template #footer>Footer content</template>
+ * </modal>
+ *
+ * Clicking outside the modal will emit the event modalMaskClicked.
+ * @example
+ * <modal @modal-mask-clicked="closeModal"></modal>
+ */
```

Comment:
I think it will be better to emit a `close` event, because that's what we want to do when a user click on it.
And we can emit the same event with a close button on the modal in the footer.

Reply:
I considered this but the modal does not control its own visibility and does not actually close when you click on the mask; that is controlled by the parent. So it would be misleading to name the event 'close' because really, the user of the modal can do whatever they want with that event.